tika-parsers/src/main/java/org/apache/tika/parser/recognition/ObjectRecognitionParser.java

```diff
@@ -83,12 +83,6 @@ public int compare(RecognisedObject o1, RecognisedObject o2) {
                 }
             };
 
-    @Field
-    private double minConfidence = 0.05;
```

Comment:
Yeah, I have moved minConfidence logic to REST servers, it is kind of odd to ask for topk objects from the backend and filter those objects again in the client with related to minConfidence and select topN objects, just too much logic in the client. we can directly ask the backend to give us topN objects which has a confidence greater than the minConfidence, less iterations and simplified client :100:

Reply:
Hey! good catch..it's not easy maintaining comments like these(https://github.com/ThejanW/tika/blob/92c65e0a43e7f09a0566bec34f352314dffe5def/tika-parsers/src/main/java/org/apache/tika/parser/recognition/ObjectRecognitionParser.java#L49-L70) 

A future developers will also miss these. Will update them asap  :+1:

Reply:
@ThejanW For my understanding `minConfidence` and `topN` can still be tweaked through Tika config / CLI options right?

Reply:
sorry, I misunderstood your question, the reason why I have removed minConfidence and topN from objectRecognitionParser is, objectRecognitionParser does not need to keep such client specific parameters. Those client specific fields should be in that specific client, we are just using ObjectRecognitionParser to process objects from the respective REST client and put them in the xhtml.

Reply:
please see,
https://github.com/ThejanW/tika/blob/master/tika-parsers/src/main/java/org/apache/tika/parser/captioning/tf/TensorflowRESTCaptioner.java#L77-L84

https://github.com/ThejanW/tika/blob/master/tika-parsers/src/main/java/org/apache/tika/parser/recognition/tf/TensorflowRESTRecogniser.java#L79-L86

https://github.com/ThejanW/tika/blob/master/tika-parsers/src/main/java/org/apache/tika/parser/recognition/tf/TensorflowRESTVideoRecogniser.java#L71-L72

Reply:
yes, minConfidence and topN can be set through CLI/ Tika Config since we have defined them in REST clients. In TensorflowRESTVideoRecogniser, you're extending TensorflowRESTRecogniser, that's why I have made some of the fields in TensorflowRESTRecogniser as protected(we need them there to derive apiUri and healthUri).

Reply:
@thammegowda your understanding is exactly correct.